### ISSUE TITLE
Blocking if number of connections >= number of threads in core.async executor

### ISSUE THREAD

**Issue by martinklepsch** [apa512/clj-rethinkdb#132]
```clojure
(defn repro []
  (let [mk-conn #(r/connect :host "127.0.0.1" :port 28015 :db "test")
        n     (-> (Runtime/getRuntime)
                  (.availableProcessors)
                  (* 2)
                  (+ 42))
        conns (doall (map (fn [_] (mk-conn)) (range n)))]
    (rethinkdb.core/close (rand-nth conns))))
```

`n` is the same as the number of threads used in the [core.async executor](https://github.com/clojure/core.async/blob/master/src/main/clojure/clojure/core/async/impl/exec/threadpool.clj#L18-L21)

Just opening this to track & in case someone else has a quick idea otherwise I'll likely get to the bottom of this later.

**Comment by danielcompton** [comment]
I think ghadi had a good idea of doing the blocking work in a `thread`. Happy to take a PR for that, or alternatively one which rips out all of the Clojure connection handling and replaces it with the Java driver.

**Comment by apa512** [comment]
This shoudn't be a problem with the new aleph code, which is merged into master and deployed as 0.14.5-SNAPSHOT.
